src/libraries/Common/src/System/Globalization/FormatProvider.Number.cs

```diff
@@ -356,7 +356,54 @@ private static unsafe bool AllowHyphenDuringParsing(NumberFormatInfo info)
                 return ret;
             }
 
+            private interface IDigitValidator
+            {
+                static abstract bool IsValidChar(char c);
+                static abstract bool IsHexBinary();
+            }
+
+            private readonly struct IntegerDigitValidator : IDigitValidator
+            {
+                public static bool IsValidChar(char c) => char.IsAsciiDigit(c);
+
+                public static bool IsHexBinary() => false;
```

Comment:
nit: `IsHexOrBinary`

-or- maybe `IsHexOrBinaryParser` and rename `IDigitValidator` to `IDigitParser`, to more closely match the other.

Reply:
Longer term I'd like to see this move even closer to the `IHexOrBinaryParser` logic we have in corelib, but that can be done in a future PR and isn't necessary as part of this.

Having the names and general logic mostly start to line up helps with that, however.

Reply:
Fixed, thanks.